modules/azure/sql.go

```diff
@@ -0,0 +1,185 @@
+	// Create a SQl Server client
+	sqlClient, err := GetSQLServerClient(subscriptionID)
+	if err != nil {
+		return "", err
+	}
+
+	// Get the corresponding server client
+	sqlServer, err := sqlClient.Get(context.Background(), resGroupName, serverName)
+	if err != nil {
+		return "", err
+	}
+
+	//Return server full domain name
+	return *sqlServer.FullyQualifiedDomainName, nil
+}
+
+// GetSQLServerState is a helper function that gets the sql server state
+func GetSQLServerState(t testing.TestingT, resGroupName string, serverName string, subscriptionID string) sql.ServerState {
+	serverState, err := GetSQLServerStateE(t, subscriptionID, resGroupName, serverName)
+	require.NoError(t, err)
+
+	return serverState
+}
+
+// GetSQLServerStateE is a helper function that gets the sql server state
+func GetSQLServerStateE(t testing.TestingT, subscriptionID string, resGroupName string, serverName string) (sql.ServerState, error) {
+	// Create a SQl Server client
+	sqlClient, err := GetSQLServerClient(subscriptionID)
+	if err != nil {
+		return "", err
+	}
+
+	// Get the corresponding server client
+	sqlServer, err := sqlClient.Get(context.Background(), resGroupName, serverName)
+	if err != nil {
+		return "", err
+	}
+
+	//Return server state
+	return sqlServer.State, nil
+}
+
+// GetDatabaseClient  is a helper function that will setup a sql DB client
+func GetDatabaseClient(subscriptionID string) (*sql.DatabasesClient, error) {
+	// Validate Azure subscription ID
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a sql DB client
+	sqlDBClient := sql.NewDatabasesClient(subscriptionID)
+
+	// Create an authorizer
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+
+	// Attach authorizer to the client
+	sqlDBClient.Authorizer = *authorizer
+
+	return &sqlDBClient, nil
+}
+
+// GetDatabaseID is a helper function that gets the sql db id
+func GetDatabaseID(t testing.TestingT, resGroupName string, serverName string, dbName string, subscriptionID string) string {
+	dbID, err := GetDatabaseIDE(t, subscriptionID, resGroupName, serverName, dbName)
+	require.NoError(t, err)
+
+	return dbID
+}
+
+// GetDatabaseIDE is a helper function that gets the sql db id
+func GetDatabaseIDE(t testing.TestingT, subscriptionID string, resGroupName string, serverName string, dbName string) (string, error) {
+	// Create a SQl db client
+	sqlClient, err := GetDatabaseClient(subscriptionID)
+	if err != nil {
+		return "", err
+	}
+
+	//Get the corresponding DB client
+	sqlDb, err := sqlClient.Get(context.Background(), resGroupName, serverName, dbName, "")
+	if err != nil {
+		return "", err
+	}
+
+	// Return DB ID
+	return *sqlDb.ID, nil
+}
+
+// GetDatabaseStatus is a helper function that gets the sql db state
+func GetDatabaseStatus(t testing.TestingT, resGroupName string, serverName string, dbName string, subscriptionID string) string {
```

Comment:
Add a method to get the SQL Server IP address.

Reply:
Azure sql servers have no static iIP but a dynamic IP that changes all the time. thus connections to them rely on using the full domain name.  there is a method that get the full domain name for sql server in the module. Also looking at the properties of the server properties , it has no way to check/retrieve the current server IP